File: anyline-tiretread-react-native/android/src/main/java/com/anyline/ttr/reactnative/CameraDirectionHelper.kt
package com.anyline.ttr.reactnative

import android.app.Activity
import android.content.Context
import android.os.Build
import android.view.Surface

object CameraDirectionHelper {

    enum class CameraDirection {
        LEFT, RIGHT, UNKNOWN
    }

    /**
     * Get current camera direction based on device orientation
     * Returns: LEFT when camera points left, RIGHT when camera points right
     */
    fun getCameraDirection(context: Context): CameraDirection {
        val rotation = if (Build.VERSION.SDK_INT >= 30) {
            context.display?.rotation
        } else {
            @Suppress("DEPRECATION")
            (context as? Activity)?.windowManager?.defaultDisplay?.rotation
        } ?: return CameraDirection.UNKNOWN

        return when (rotation) {
            Surface.ROTATION_90 -> CameraDirection.LEFT    // Camera points left
            Surface.ROTATION_270 -> CameraDirection.RIGHT  // Camera points right
            else -> CameraDirection.UNKNOWN                // Portrait or unknown
        }
    }
}

File: anyline-tiretread-react-native/ios/CameraDirectionHelper.swift
//
//  CameraDirectionHelper.swift
//  anyline-ttr-react-native
//
//  Helper class to determine camera direction based on device orientation
//

import UIKit

enum CameraDirection {
    case left
    case right
    case unknown
}

class CameraDirectionHelper {
    /**
     * Get current camera direction based on interface orientation
     * Returns: .left when camera points left, .right when camera points right
     */
    static func getCameraDirection() -> CameraDirection {
        let orientation = UIApplication.shared.statusBarOrientation

        switch orientation {
        case .landscapeLeft:
            return .right  // Camera points right
        case .landscapeRight:
            return .left   // Camera points left
        default:
            return .unknown // Portrait or unknown
        }
    }

    /**
     * Convert CameraDirection to string for JavaScript bridge
     */
    static func cameraDirectionToString(_ direction: CameraDirection) -> String {
        switch direction {
        case .left:
            return "LEFT"
        case .right:
            return "RIGHT"
        case .unknown:
            return "UNKNOWN"
        }
    }
}